lib/compat/wordpress-6.5/html-api/class-gutenberg-html-tag-processor-6-5.php

```diff
@@ -1651,7 +1651,7 @@ private function apply_attributes_updates( $shift_this_point = 0 ) {
 		 * replacements adjust offsets in the input document.
 		 */
 		foreach ( $this->bookmarks as $bookmark_name => $bookmark ) {
-			$bookmark_end   = $bookmark->start + $bookmark->length;
+			$bookmark_end = $bookmark->start + $bookmark->length;
```

Comment:
This change is out-of-scope for this PR. Need to do it in a separate issue/PR please.

Reply:
I would say it's related to **_some degree_** because the following exclude pattern was added to `phpcs.xml.dist` to prevent these files from being checked against the `WordPress-Docs` ruleset:

```xml
<exclude-pattern>./lib/compat/wordpress-*/html-api/*.php</exclude-pattern>
```
If `WordPress-Docs` is disabled, the `<exclude-pattern>` expression above is no longer needed.
However, I don't mind submitting another pull request to address that specific issue. I will revert this now.

Reply:
Fixed in ae3f74476b4ce9bddcc82d9e2572826650624e4d.

Reply:
I see this out of scope because the alignment of `=` in the source code is not part of the `WordPress-Docs` ruleset.

Reply:
I'd suggest making a PR for that alignment issue and then once merged into `trunk`, then rebase this PR to include that change.

Reply:
I've created a PR for the alignment issue. Could you please approve it, @hellofromtonya? 
https://github.com/WordPress/gutenberg/pull/57030

Reply:
Okay @anton-vlasenko that PR is merged and ready to bring into this PR.

Reply:
Thank you, @hellofromtonya!
I've rebased this PR.

Reply:
as noted in the linked PR, this exclude ruleset exists in part because Gutenberg is _not_ the place to address code changes to the HTML API, as the files in Gutenberg are largely built artifacts from Core, other than renaming the classes to `Gutenberg____6_5` and all.

just as we don't link the minified built JS files, we need to make changes in Core first and then backport them into Gutenberg once they are reviewed, accepted, and merged.

Reply:
was there a plan to add the rule back in that was removed for the html api? this caused another hassle with @SergeyBiryukov's change that started blocking backports into Gutenberg.

was it even intentional that the HTML API exclusion was removed?

https://github.com/WordPress/gutenberg/commit/b592ba99881818ed52b7d05ae9614e8ab69aa45e#diff-05ae9cddcaec1e845771a7db224961439f83ef5939ec67d3a48744cb34d7e58bL62-L63

should we add that exclusion back in?

Reply:
> was it even intentional that the HTML API exclusion was removed?

@dmsnell 
Yes, it was 100% intentional.
Personally, I'd only add exceptions for conflicting rules, like `VariableAnalysis.CodeAnalysis.VariableAnalysis`.
However, if it makes things easier, please add it back.

Reply:
thanks. I'll do that and add you as a reviewer. given how frequently it seems that the rules change, and how people rush through changes without review that end up introducing challenges I'd rather we treat the HTML API compatibility layer just like we treat NPM packages, which is why we added this exclusion originally.

thanks for your guidance on this.

is there anything we can do to alert someone not to remove the rule again? I originally thought that its very nature of existing as an exclusion would prevent it, but would a large "DO NOT REMOVE" warning help? I prefer to avoid those if possible.

Reply:
thanks. I'll try and be more obtuse in the language. When this was originally removed [in b592ba99](https://github.com/WordPress/gutenberg/commit/b592ba99881818ed52b7d05ae9614e8ab69aa45e) it read…

```xml
<!-- Exclude files maintained in WordPress Core and backported to Gutenberg -->
```

I've updated it in #57922 to read…

```xml
<!-- Exclude files maintained in WordPress Core and backported to Gutenberg
	     DO NOT REMOVE these rules; these files are "built" artifacts from Core
	     and when they are removed it prevents keeping the repos in sync. -->
```

Hoping this does a better job.